### ISSUE TITLE
Work towards mocking all the online tests in SunPy

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#2874]

**Comment by Cadair** [comment]
I believe @Chicrala is taking `sunpy/net/dataretriever/test_norh.py` and `sunpy/net/dataretriever/test_fermi.py`

**Comment by Cadair** [comment]
I generated this report on the tests which are covered by our online tests but not our offline ones: https://pste.eu/p/pXZS.html

**Comment by Chicrala** [comment]
@Cadair  Is this link broken or it actually should display an error "404 not found."?

**Comment by Cadair** [comment]
boo it must have deleted it after some amount of time :woman_facepalming:

**Comment by yashrsharma44** [comment]
Hey, I was interested in working some of the portions. Can someone explain the progress, and help me out with the parts that I can work upon?

**Comment by nabobalis** [comment]
There is not a fixed process. The idea is to replicate any test that has a `@pytest.mark.remote_data` decorator  and use the mock library (linked in the original post) to replace the online response with a offline version.

**Comment by Cadair** [comment]
Also for the record, it's worth pointing out that there should still be some online tests to detect changes to the remote services etc. The objective of this issue is to ensure that as much of the sunpy code is tested offline as possible.

**Comment by Chicrala** [comment]
@Cadair, I was discussing that with @nabobalis and we were wondering that maybe the best approach would be to ''duplicate'' the tests with an offline version of them.

**Comment by Cadair** [comment]
The best thing to do is to test the mocks, i.e. if you mock the response to an API call, you want a test to check that that response dosen't change.

**Comment by darthgera123** [comment]
are all tests mocked or are some left?

**Comment by Chicrala** [comment]
Do you want to do fermi? I'm still struggling with norh.

**Comment by darthgera123** [comment]
cool ill do fermi

**Comment by nabobalis** [comment]
test_norh was is "done" here #2923

**Comment by Chicrala** [comment]
Yeah that's the guy who's currently helping me to understand how to make those tests. I was struggling too much to figure it out. So norh is closed?

**Comment by nabobalis** [comment]
Not yet, the pull request is not merged in. It needs another review but it will be assuming that PR is ready.

**Comment by nabobalis** [comment]
I will try to keep track of the PRs sent in for this issue and update the main post.

**Comment by darthgera123** [comment]
fermi is open right?

**Comment by nabobalis** [comment]
It should be.

**Comment by darthgera123** [comment]
@yashrsharma44  im almost done with mocking test_fermi

**Comment by yashrsharma44** [comment]
Hey, @darthgera , go ahead and make a PR, I haven't been working on that.
BTW, you could have made a PR without pinging me 😛

**Comment by darthgera123** [comment]
i guess fermi is completed so if there is any other that needs to be mocked please tell

**Comment by jeffreypaul15** [comment]
Is this still a requirement?
If yes, is it still mock and the pytest-mock plugin that is required?

**Comment by nabobalis** [comment]
Yes, whatever works.

**Comment by wtbarnes** [comment]
This issue needs to be broken out into sub-issues for each client still needing mocking.
